src/Language/Bitcoin/Script/Descriptors/Parser.hs

```diff
@@ -37,6 +41,13 @@ import Language.Bitcoin.Utils (
     maybeFail,
  )
 
+
+parseChecksumDescriptor :: Network -> Text -> Either String ChecksumDescriptor
```

Comment:
I am OK with just calling this function `parseChecksumDescriptor`, even though it is a breaking change.  Now that we have your logic for dealing with checksums, I don't think anyone would want to use the old `parseDescriptor`.

Reply:
Oops, I'm trying to say we just make this `parseDescriptor` and not have a separate checksum-unaware version.

Reply:
Makes sense. Was trying not to break the exported functions until I was sure that's what you had in mind.